packages/vscode-extension/src/project/project.ts

```diff
@@ -98,6 +90,18 @@ export class Project implements Disposable, ProjectInterface, DeviceSessionsMana
       }
     );
 
+    this.projectState = {
```

Comment:
this will likely conflict with #1154

Reply:
Yes, but we have to pass it somehow, and this seems to be the most sensible place to me.
It shouldn't be hard to adapt either PR to this change.